guides/image-sources/provisioning-image-cloudinit.plantuml

```diff
@@ -0,0 +1,55 @@
+@startuml
+
+!include foreman.pstyle
+
+title Image-based provisioning with cloud-init configuration
+
+actor       User
+participant "Provisioned\nInstance" as Host
+participant "Foreman" as Foreman
+participant "Foreman\nProxy" as Proxy
+participant "Infrastructure\nCloud" as Cloud
+participant DNS
+!if ($puppet)
+participant "Puppet\nserver" as Puppet
+!endif
+
+note over Foreman : has an image\nwith cloud-init flag
+
+== Create host in Foreman ==
+
+User -> Foreman : clicks **Create Host**
+User -> Foreman : selects cloud resource,\nthe image and submits
+group Template [cloud-init]
+   Foreman -> Proxy : renders the cloud-init script
+end
+Foreman -> Cloud : creates new instance\nwith cloud-init info
+Cloud -> Foreman : reports IP address
+Foreman -> Proxy : creates DNS records
+Proxy -> DNS : forwards DNS records
```

Comment:
This is not entirely correct. First the code reference: https://github.com/theforeman/foreman/blob/e80d5558767bb713c13264e821122baea577af76/app/models/concerns/orchestration/dns.rb#L63-L71

It will create all feasible DNS record types. That can mean `A`, `AAAA` and `PTR` (for both v4 and v6), depending on the known IP addresses. So the first part is simplified, but correct.

The second part is IMHO confusing. The Proxy tells the DNS server to create the DNS records. A Foreman Proxy, especially the DNS module, implements the [facade pattern](https://en.wikipedia.org/wiki/Facade_pattern). It abstracts away the actual implementation away from Foreman. IMHO it is forwarding the DNS record creation, not the records themselves.

Reply:
So what wording do you suggest precisely?